### DISCUSSION TITLE
ebusd on Raspberry Pi 5

### DISCUSSION THREAD

**Discussion by urzeitfurz** [john30/ebusd#1101] [Q&A]
Hi,
I have a eBUS Adapter v5 and tried it with a Raspberry Pi 4 Model B. It works very well. Now I bougt a Raspberry Pi 5 and I'm not able to make it work.

`ls -l /dev`
shows me that serial0 belongs to ttyAMA10. So I adapt it to my ebusd-configfile with
`--device=ens:/dev/ttyAMA10`

At the Pi 5 uart should always be enabled. So I added nothing to /boot/config.txt.

`dmesg|grep ttyAMA10` shows me this:
`[    0.011049] 107d001000.serial: ttyAMA10 at MMIO 0x107d001000 (irq = 15, base_baud = 0) is a PL011 rev2`
I think its fine.

`pinctrl get 14-15` says
`14: a4    pn | hi // PIN8/GPIO14 = TXD0`
`15: a4    pu | hi // PIN10/GPIO15 = RXD0`

`ebusctl state` says "no signal". So I'm not able to read any value from the ebus.

I don't know where the problem is. When I put the adapter back on the Pi 4 everythings works fine.

Does someone have an idea where the problem is or runs it already on a Pi 5?

Thank you.

**Comment by urzeitfurz** [comment]
UPDATE

The adapter works now on the Pi 5. I connected it via USB. When I connect it over the GPIOs it doesn't work.

There is one more thing I established today.
On the Pi 4 there is the fading of the LEDs of the ebus-controller while starting. Then the LED are blinking.
On the Pi 5 is no fading. When I press the button on the controller the blue LED turns on. After that there is no LED shining. I also set the mode of the controller to "raspberry pi" with the wifi web interface.

**Comment by krusty-is-cool** [comment]
Hi,

I have the same issue and I also used USB instead.

I found [this thread](https://forums.raspberrypi.com/viewtopic.php?t=359132&sid=bc01aaca27b62d938bbe1a4637fa1c0a&start=25) about UART on the Raspberry Pi 5, where a Raspberry Pi engineer explained how to setup UART0/ttyAMA0 on GPIOs 14 & 15:

> dtparam=uart0 just does the first part, i.e. enabling UART0/ttyAMA0 on GPIOs 14 & 15.

> Does clicking on Enable Serial in settings still set /dev/serial0 to pins 14/15 on a Pi5 like every other Pi before it did since Stretch.

> Yes, by adding "dtparam=uart0" to config.txt.

But I couldn't make it work. I am new to Raspberry Pi (and ebusd) though, I might have done something wrong.
